binding.cc

```diff
@@ -514,6 +534,414 @@ bare_sdl_is_audio_device_paused(
   return SDL_AudioDevicePaused((SDL_AudioDeviceID) deviceId);
 }
 
+static std::vector<uint32_t>
+bare_sdl_get_cameras(
+  js_env_t *env,
+  js_receiver_t
+) {
+  int count = 0;
+  SDL_CameraID *devices = SDL_GetCameras(&count);
+
+  std::vector<uint32_t> list;
+
+  if (devices != nullptr) {
+    for (int i = 0; i < count; i++) {
+      list.push_back(devices[i]);
+    }
+    SDL_free(devices);
+  }
+
+  return list;
+}
+
+static std::optional<std::string>
+bare_sdl_get_camera_name(
+  js_env_t *env,
+  js_receiver_t,
+  uint32_t device_id
+) {
+  const char *name = SDL_GetCameraName((SDL_CameraID)device_id);
+  return name ? std::optional<std::string>(name) : std::nullopt;
+}
+
+static uint32_t
+bare_sdl_get_camera_position(
+  js_env_t *env,
+  js_receiver_t,
+  uint32_t device_id
+) {
+  return (uint32_t)SDL_GetCameraPosition((SDL_CameraID)device_id);
+}
+
+static std::vector<js_arraybuffer_t>
+bare_sdl_get_camera_supported_formats(
+  js_env_t *env,
+  js_receiver_t,
+  uint32_t device_id
+) {
+  int err;
+  int count = 0;
+  SDL_CameraSpec **specs = SDL_GetCameraSupportedFormats((SDL_CameraID)device_id, &count);
+
+  std::vector<js_arraybuffer_t> list;
+
+  if (specs != nullptr) {
+    for (int i = 0; i < count; i++) {
+      if (specs[i]) {
+        js_arraybuffer_t handle;
+        bare_sdl_camera_format_t *format;
+        err = js_create_arraybuffer(env, format, handle);
+        assert(err == 0);
+
+        format->spec = *specs[i];
+        format->index = i;
```

Comment:
We do we keep a trace of the `index` here?

Reply:
Ooops! Yeah, that wasn't needed.